### PR TITLE
feat: implement modal form

### PR DESCRIPTION
**Pull request by Lanng21**
image
image

### CODE REVIEW

File: src/controllers/bookController.js [hunk 1]
```diff
@@ -13,39 +13,46 @@ class BookController {
     this.formView.setOnSubmit(this.handleFormSubmit.bind(this));
     this.listView.setOnEdit(this.handleEdit.bind(this));
     this.listView.setOnDelete(this.handleDelete.bind(this));
+
+    this.createBookButton = document.querySelector('.create-book');
```
Comment (baonguyen26) [inline src/controllers/bookController.js:17]: Do not handle this in the controller, It should be handled in view

File: src/controllers/bookController.js [hunk 2]
```diff
@@ -13,39 +13,46 @@ class BookController {
...
+      this.handleCreateBookButtonClick.bind(this),
+    );
+  }
+
+  handleCreateBookButtonClick() {
+    this.formView.render();
   }
 
-  async handleFormSubmit(title, author, genre, publishedYear) {
+  showSuccessMessage(message) {
```
Comment (baonguyen26) [inline src/controllers/bookController.js:28]: same

File: src/controllers/bookController.js [hunk 3]
```diff
@@ -13,39 +13,46 @@ class BookController {
...
+    // Show the success message using the toast
+    this.toast.show(message, 'success');
+  }
+
+  async handleFormSubmit(title, author, category, publishedYear) {
     try {
       if (this.model.currentBook) {
-        await this.updateBook(title, author, genre, publishedYear);
+        await this.updateBook(title, author, category, publishedYear);
```
Comment (baonguyen26) [inline src/controllers/bookController.js:36]: Im not sure about this, you wanna update but do you not need an id?

File: src/controllers/bookController.js [hunk 4]
```diff
@@ -13,39 +13,46 @@ class BookController {
...
       } else {
-        await this.addBook(title, author, genre, publishedYear);
+        await this.addBook(title, author, category, publishedYear);
       }
 
       await this.fetchBooks();
       this.formView.clearForm();
       this.formView.showSuccessMessage('Book saved successfully.');
```
Comment (baonguyen26) [inline src/controllers/bookController.js:43]: move to view

File: src/controllers/bookController.js [hunk 5]
```diff
@@ -13,39 +13,46 @@ class BookController {
...
-      this.model.currentBook = null; // Reset currentBook after successful update or add
-      this.formView.render(); // Render the form with the updated state (Add button)
+      this.model.currentBook;
+      this.formView.render();
     } catch (error) {
       this.formView.showErrorMessage('Error saving book. Please try again.');
```
Comment (baonguyen26) [inline src/controllers/bookController.js:47]: and here

File: src/models/book.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
 class Book {
-  constructor(id, title, author, genre, publishedYear) {
+  constructor(id, title, author, category, publishedYear) {
     this.id = id;
```
Comment (baonguyen26) [inline src/models/book.js:3]: Do you need an id?

File: src/models/bookModel.js [hunk 7]
```diff
@@ -7,16 +7,10 @@ class BookModel {
   }
 
   setBooks(books) {
-    this.books = books.map(
-      (bookData) =>
-        new Book(
-          bookData.id,
```
Comment (baonguyen26) [inline src/models/bookModel.js:13]: Do you need?

**Review by baonguyen26 — CHANGES_REQUESTED** [review]
Pls update this PR and split file changes, I can't review it for you

File: src/controllers/bookController.js (hunk 3 above)
Comment (Lanng21) [reply src/controllers/bookController.js:36]: i will get id on bookModel and in controller Im call `this.model.curentBook` and update.
```
  updateBook(updatedBook) {
    const index = this.books.findIndex((book) => book.id === updatedBook.id);
    if (index !== -1) {
      const { id, title, author, category, publishedYear } = updatedBook;
      this.books[index] = new Book(id, title, author, category, publishedYear);
    }
  }
```
this is updateBook on bookmodel

File: src/models/book.js (hunk 6 above)
Comment (Lanng21) [reply src/models/book.js:3]: yes, i need

**Pull request closed without merging**